pkg/render/compliance.go

```diff
@@ -1769,3 +1829,125 @@ func (c *complianceComponent) complianceServerAllowTigeraNetworkPolicy() *v3.Net
+					// request down the tunnel
+					"get",
+				},
+			},
+		},
+	})
+
+	// In a single tenant setup we want to create a cluster role that binds using service account
+	// tigera-compliance-server from tigera-compliance namespace. In a multi-tenant setup
+	// Compliance server from the tenant's namespace impersonates service tigera-compliance-server
+	// from tigera-compliance namespace
+	objects = append(objects, &rbacv1.ClusterRoleBinding{
+		TypeMeta:   metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: MultiTenantComplianceManagedClustersAccessClusterRoleName},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     MultiTenantComplianceManagedClustersAccessClusterRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			// requests for policy recommendation to managed clusters are done using service account tigera-compliance-server
+			// from tigera-compliance namespace regardless of tenancy mode (single tenant or multi-tenant)
+			{
+				Kind:      "ServiceAccount",
+				Name:      ComplianceServerServiceAccount,
+				Namespace: ComplianceNamespace,
+			},
+		},
+	})
+
+	return objects
+}
+
+func (c *complianceComponent) multiTenantComplianceSnapshotterAccess() []client.Object {
+	var objects []client.Object
+	objects = append(objects, &rbacv1.ClusterRole{
+		TypeMeta:   metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: ComplianceSnapshotterServiceAccount},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"linseed.tigera.io"},
+				Resources: []string{"snapshots"},
+				Verbs:     []string{"get", "create"},
+			},
+		},
+	})
+
+	objects = append(objects, c.complianceSnapshotterClusterRoleBinding())
+
+	return objects
+}
+
+func (c *complianceComponent) multiTenantComplianceBenchmarkerAccess() []client.Object {
+	var objects []client.Object
+	objects = append(objects, &rbacv1.ClusterRole{
+		TypeMeta:   metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: ComplianceBenchmarkerServiceAccount},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"linseed.tigera.io"},
+				Resources: []string{"benchmarks"},
+				Verbs:     []string{"get", "create"},
+			},
+		},
+	})
+
+	objects = append(objects, c.complianceBenchmarkerClusterRoleBinding())
+
+	return objects
+}
+
+func (c *complianceComponent) multiTenantComplianceControllerAccess() []client.Object {
+	var objects []client.Object
+	objects = append(objects, &rbacv1.ClusterRole{
+		TypeMeta:   metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: ComplianceControllerServiceAccount},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"linseed.tigera.io"},
+				Resources: []string{"compliancereports"},
```

Comment:
I don't understand how this is laid out. Maybe some code comments would help here, but it looks like this permission should already be granted via `complianceControllerClusterRole` - why are we not using that permission here?

Same question for the other mirror functions

Reply:
Because those have additional rights to get pods, node etc needed for reports on the management cluster or managed cluster. I was thinking we can provide only linseed RBAC.